core/app/models/spree/order_inventory.rb

```diff
@@ -53,12 +53,12 @@ def remove(item_units, shipment = nil)
     # first unshipped that already includes this variant
     # first unshipped that's leaving from a stock_location that stocks this variant
     def determine_target_shipment
-      shipment = order.shipments.detect do |shipment|
-        shipment.ready_or_pending? && shipment.include?(variant)
+      shipment = order.shipments.detect do |s|
+        s.ready_or_pending? && s.include?(variant)
```

Comment:
This is also variable shadowing, and I think here I'd be happier if the variable outside the block would be something like  `pending_shipment_with_variant`. Or extract it into a method with that name.